sklearn/preprocessing/tests/test_data.py

```diff
@@ -1498,3 +1498,25 @@ def test_one_hot_encoder_unknown_transform():
     oh = OneHotEncoder(handle_unknown='42')
     oh.fit(X)
     assert_raises(ValueError, oh.transform, y)
+
+
+def test_fit_cold_start():
+    X = iris.data
+    X_2d = X[:, :2]
+
+    # Scalers that have a partial_fit method
+    scalers = [StandardScaler(with_mean=False, with_std=False),
+               MinMaxScaler(),
+               MaxAbsScaler()]
+
+    for scaler in scalers:
+        scaler.fit_transform(X)
+        # with a different shape, this may break the scaler unless the internal
+        # state is reset
+        try:
+            scaler.fit_transform(X_2d)
+        except ValueError as err:
+            print("Cannot fit %s a second time with different shape "
+                  "of input. Error message: %s"
+                  % (scaler.__class__.__name__, str(err)))
+            assert False
```

Comment:
Maybe you could leave it as a smoke test.

```python
for scaler in scalers:
    scaler.fit_transform(X)
    # with a different shape, this may break the scaler unless the internal
    # state is reset
    scaler.fit_transform(X_2d)
```

I am not sure what your try/except gives you. In particular, stdout is hidden by default in nosetests and even if it was not hidden it would very likely be not very obvious to spot because of the exception stack created by `assert False`.